baseapp/abci.go

```diff
@@ -317,6 +323,51 @@ func (app *BaseApp) Query(req abci.RequestQuery) abci.ResponseQuery {
 	return sdkerrors.QueryResult(sdkerrors.Wrap(sdkerrors.ErrUnknownRequest, "unknown query path"))
 }
 
+func handleQueryGRPC(app *BaseApp, handler GRPCQueryHandler, req abci.RequestQuery) abci.ResponseQuery {
+	// when a client did not provide a query height, manually inject the latest
+	if req.Height == 0 {
+		req.Height = app.LastBlockHeight()
+	}
+
+	if req.Height <= 1 && req.Prove {
+		return sdkerrors.QueryResult(
+			sdkerrors.Wrap(
+				sdkerrors.ErrInvalidRequest,
+				"cannot query with proof when height <= 1; please provide a valid height",
+			),
+		)
+	}
+
+	cacheMS, err := app.cms.CacheMultiStoreWithVersion(req.Height)
+	if err != nil {
+		return sdkerrors.QueryResult(
+			sdkerrors.Wrapf(
+				sdkerrors.ErrInvalidRequest,
+				"failed to load state at height %d; %s (latest height: %d)", req.Height, err, app.LastBlockHeight(),
+			),
+		)
+	}
+
+	// cache wrap the commit-multistore for safety
+	ctx := sdk.NewContext(
+		cacheMS, app.checkState.ctx.BlockHeader(), true, app.logger,
+	).WithMinGasPrices(app.minGasPrices)
+
+	res, err := handler(ctx, req)
+
+	if err == nil {
+		return res
+	}
+
+	space, code, log := sdkerrors.ABCIInfo(err, false)
+	return abci.ResponseQuery{
+		Code:      code,
+		Codespace: space,
+		Log:       log,
+		Height:    req.Height,
+	}
+}
```

Comment:
This should more like `handleQueryCustom`, especially since `if err != nil {}` is more common:

```go
	if err != nil {
		space, code, log := sdkerrors.ABCIInfo(err, false)
		return abci.ResponseQuery{
			Code:      code,
			Codespace: space,
			Log:       log,
			Height:    req.Height,
		}
	}

	res
```

Reply:
@alexanderbez I suggested the change and perhaps am preaching to the choir but the current change is a more idiomatic Go style for early returns.

Reply:
I thought I had just copied from `handleQueryCustom`, not sure how that got inverted

Reply:
All good, not blocking here 👍 

consistency > everything IMHO